website/main/content/blog/2025-04-07-merged-farming/index.md

```diff
@@ -0,0 +1,97 @@
+
+I actually explored several solutions for this throughout the week:
+- The first one is a bit complex and may have been an overkill in hindsight (thank you Nazar for pointing this out). I won't jump into the nitty gritty details and will leave this image explaining the high-level here for reference. The idea was for every parent shard to build a tree with all the segments from their children. With every new segment created in the parent chain an updated commitment for the tree of children segments is attached to the parent shard segment along with proofs for all the new segments included in the tree since the last update. In this way, parent chain segments implicitly propagate relevant information about their children to their upper layers (and eventually the beacon chain).
+
+<p align="center">
+<img alt="Hierarchical Verkle Tree of Segments" src="hierarchical_verkle_trees.png"></img>
+</p>
+
+- Nazar rightly pointed out the following: _why wait for parent chain segment generation as the cut-off point for committing segments? Why not batch whatever segments from the children have been submitted to a shard to propagate them to upper layers as soon as they have been included in a block?_ And this is the solution that I will consider for the design. The process is as follows:
+- Let's take as an example the tree of shards from the figure above. Shards C and D are generating new segments and submitting their commitments (or segment headers) to their parent, A.
+- These segments are submitted to A by nodes from C and D as transactions that will be included in a block of A.
+- As soon as A verifies a block including transactions with segments from C and D, nodes in A will create a "super segment" commitment (this is how we are going to call this data structure), that creates a commitment for all the segments included in the block of A. 
+- Nodes in A will be periodically creating their own segments and committing them to the beacon chain, as well as the super segment commitment for the segments from C and D triggered through the submission of segments in A.
+- The super segment commitment will be a vector commitment that proves the commitment of segments in the history of A, and are used to include segments from deeper layer in batch into the global history kept in the beacon chain. Super segments will be sequenced along with regular segments from immediate children of the beacon chain conforming the global history of the system with all of the segments created in all shards.
```

Comment:
I think it'd be helpful to mention that inclusion on the beacon chain is what gives segments their order in the global history. As well as mentioning that pieces will have an extended information in them to verify this stuff.

This feels like a missing piece for a complete implementation without additional diagrams.

Reply:
Added a few more bullet points with more details about this